cai_causal_graph/graph_components.py

```diff
@@ -292,24 +317,109 @@ def variable_name(self) -> str:
             raise ValueError(f'The variable name for node {self.identifier} is not set.')
         return name
 
-    def __eq__(self, other: object) -> bool:
+    def __eq__(self, other: object, deep: bool = False) -> bool:
         """
         Check if a node is equal to another node.
 
         This method checks for the node identifier, variable name, and time lag, but ignores variable type,
         inbound/outbound edges, and any other metadata.
+
+        :param other: The other node to compare with.
+        :param deep: If True, then the variable type, inbound/outbound edges, and metadata are also checked.
         """
         if not isinstance(other, TimeSeriesNode):
             return False
         # As TimeSeriesNode is a subclass of Node, we can use the Node.__eq__ method and then check for the properties
-        if not super().__eq__(other):
+        if not super().__eq__(other, deep):
             return False
         return self.variable_name == other.variable_name and self.time_lag == other.time_lag
 
     def __hash__(self) -> int:
         """Return a hash value of the node identifier."""
         return super().__hash__()
 
+    def to_dict(self, include_meta: bool = True) -> dict:
+        """
+        Serialize a `cai_causal_graph.graph_components.TimeSeriesNode` instance to a dictionary.
+
+        :param include_meta: Whether to include meta information about the node in the dictionary. Default is `True`.
+        :return: The dictionary representation of the `cai_causal_graph.graph_components.TimeSeriesNode` instance.
+        """
+        dictionary = super().to_dict(include_meta)
+        # add the time lag and variable name to the dictionary
+        dictionary.update({TIME_LAG: self.time_lag, VARIABLE_NAME: self.variable_name})
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> TimeSeriesNode:
+        """
+        Return a `cai_causal_graph.graph_components.TimeSeriesNode` instance from a dictionary.
+
+        :param dictionary: The dictionary from which to create the time series node.
+        :return: The time series node created from the dictionary.
+        """
+        assert 'identifier' in dictionary, 'The dictionary must contain the `identifier` key.'
+
+        # check the meta for the time lag and variable name match the ones in the dictionary
+        time_lag = dictionary.get(TIME_LAG, None)
+        variable_name = dictionary.get(VARIABLE_NAME, None)
+
+        if 'meta' not in dictionary:
+            dictionary['meta'] = {}
```

Comment:
You are modifying the dictionary here. User may be confused. Would do `meta = dictionary.get('meta', {})` above. Then use `meta` throughout. Otherwise the dict outside the scope of this method is changed.